### ISSUE TITLE
DeepLink Feature Request

### ISSUE THREAD

**Issue by metallica1375** [Mercen-Lee/FlowKit#5]
hey can you implement the Deeplink in FlowKit?

**Comment by mercen-lee** [comment]
You mean, like [`onOpenURL(perform:)`](https://developer.apple.com/documentation/swiftui/view/onopenurl(perform:)) in SwiftUI?

**Comment by metallica1375** [comment]
yes, some sort of that, or even a better solution if you know, because for navigation to a new page it requires to instantiate a View Struct and pass it to the flow.push()

**Comment by mercen-lee** [comment]
Resolved in 86cdb21 on main.

Added `FlowDeepLink` plus `registerDeepLink`, `registerDeepLinks`, `removeDeepLinks`, and `openDeepLink(_:)`. Routes can match by path, host/path, or a custom matcher and then run FlowKit navigation actions.

Validation: `xcodebuild -scheme FlowKit -destination "generic/platform=iOS Simulator" -sdk iphonesimulator build` succeeded.
